mlflow/langchain/utils/chat.py

```diff
@@ -0,0 +1,298 @@
+    index: int
+    finish_reason: Optional[str] = None
+    delta: _ChatDeltaMessage
+
+
+class _ChatUsage(pydantic.BaseModel, extra="forbid"):
+    prompt_tokens: Optional[int] = None
+    completion_tokens: Optional[int] = None
+    total_tokens: Optional[int] = None
+
+    @staticmethod
+    def get_schema():
+        return Schema(
+            [
+                ColSpec(DataType.integer, "prompt_tokens", required=False),
+                ColSpec(DataType.integer, "completion_tokens", required=False),
+                ColSpec(DataType.integer, "total_tokens", required=False),
+            ]
+        )
+
+
+class _ChatResponse(pydantic.BaseModel, extra="forbid"):
+    id: Optional[str] = None
+    object: Literal["chat.completion"] = "chat.completion"
+    created: int
+    # Make the model field optional since we may not be able to get a stable model identifier
+    # for an arbitrary LangChain model
+    model: Optional[str] = None
+    choices: List[_ChatChoice]
+    usage: _ChatUsage
+
+    @staticmethod
+    def get_schema():
+        return Schema(
+            [
+                ColSpec(DataType.string, "id", required=False),
+                ColSpec(DataType.string, "object"),
+                ColSpec(DataType.integer, "created"),
+                ColSpec(DataType.string, "model", required=False),
+                ColSpec(Array(_ChatChoice.get_schema()), "choices"),
+                ColSpec(_ChatUsage.get_schema(), "usage"),
+            ]
+        )
+
+
+class _ChatChunkResponse(pydantic.BaseModel):
+    id: Optional[str] = None
+    object: Literal["chat.completion.chunk"] = "chat.completion.chunk"
+    created: int
+    # Make the model field optional since we may not be able to get a stable model identifier
+    # for an arbitrary LangChain model
+    model: Optional[str] = None
+    choices: List[_ChatChoiceDelta]
+
+
+def try_transform_response_to_chat_format(response):
+    if isinstance(response, str):
+        message_content = response
+        message_id = None
+    elif isinstance(response, AIMessage):
+        message_content = response.content
+        message_id = getattr(response, "id", None)
+    else:
+        return response
+
+    transformed_response = _ChatResponse(
+        id=message_id,
+        created=int(time.time()),
+        model=None,
+        choices=[
+            _ChatChoice(
+                index=0,
+                message=_ChatMessage(
+                    role="assistant",
+                    content=message_content,
+                ),
+                finish_reason=None,
+            )
+        ],
+        usage=_ChatUsage(
+            prompt_tokens=None,
+            completion_tokens=None,
+            total_tokens=None,
+        ),
+    )
+
+    if IS_PYDANTIC_V1:
+        return json.loads(transformed_response.json())
+    else:
+        return transformed_response.model_dump(mode="json")
+
+
+def try_transform_response_iter_to_chat_format(chunk_iter):
```

Comment:
```suggestion
def try_transform_stream_response_to_chat_format(chunk_iter):
```

nit: to be more informative (`iter` is in param name anyway)